### ISSUE TITLE
Unable to access S3bucket from ec2-instance created through auto scaling group

### ISSUE THREAD

**Issue by zymr-keshav** [aws/aws-sdk-js#1291]
I have created a _classic load balancer_ then _launch configuration_ and _auto scaling group_ and add _scaling policy_ with _cloud watch alarm_

now when I logged in into classic load balancer through ssh and check the log file but it gives error 

```
INFO: Unable to execute HTTP request: connect timed out
java.net.SocketTimeoutException: connect timed out
    at java.net.PlainSocketImpl.socketConnect(Native Method)
    at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
    at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
    at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
    at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
    at java.net.Socket.connect(Socket.java:589)
    at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:668)
    at org.apache.http.conn.ssl.SSLSocketFactory.connectSocket(SSLSocketFactory.java:524)
    at org.apache.http.conn.ssl.SSLSocketFactory.connectSocket(SSLSocketFactory.java:403)
    at com.amazonaws.http.conn.ssl.SdkTLSSocketFactory.connectSocket(SdkTLSSocketFactory.java:131)
    at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:177)
    at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:304)
    at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:611)
    at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:446)
    at org.apache.http.impl.client.AbstractHttpClient.doExecute(AbstractHttpClient.java:863)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
    at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:57)
    at com.amazonaws.http.AmazonHttpClient.executeOneRequest(AmazonHttpClient.java:822)
    at com.amazonaws.http.AmazonHttpClient.executeHelper(AmazonHttpClient.java:576)
    at com.amazonaws.http.AmazonHttpClient.doExecute(AmazonHttpClient.java:362)
    at com.amazonaws.http.AmazonHttpClient.executeWithTimer(AmazonHttpClient.java:328)
    at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:307)
    at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:3659)
    at com.amazonaws.services.s3.AmazonS3Client.getObject(AmazonS3Client.java:1162)
    at com.amazonaws.services.s3.AmazonS3Client.getObject(AmazonS3Client.java:1051)

```

In my program , I tried to access s3 bucket through the ec2-instance and it does not work;

my s3 bucket have the valid permission for the role I have assigned to create instance.

also I have added ssh and icmp inbound rule with the security group attached with load balancer.

Kindly tell me what is the issue.

**Comment by chrisradek** [comment]
@zymr-keshav 
It looks like you're using the AWS SDK for Java, whose github can be found here:
https://github.com/aws/aws-sdk-java

Please re-open if the issue is actually with the AWS SDK for JavaScript. If so, please provide additional information, such as SDK version, node version, and some sample code that throws the error.

**Comment by zymr-keshav** [comment]
I am using AWS SDK for javascript to create load balancer, launch configuration and autoscaling policy; by that way it creates 2 instance and when I logged in through `ssh -i key.pem ec2-user@load-balancer-dns-name` and check for the log file than it gives error.  as there is a java program that run through UserData while creating instance and above is the output of that log. 

even I checked that none of the `aws-cli` command working inside the ec2 instance .  here i also posted the same query regarding the security group.
http://serverfault.com/questions/825036/can-we-use-same-security-group-in-launch-configuration-and-classic-load-balancer/825127#825127
